client/lib/analytics/signup.js

```diff
@@ -66,29 +73,15 @@ export function recordSignupInvalidStep( flow, step ) {
 	analytics.tracks.recordEvent( 'calypso_signup_goto_invalid_step', { flow, step } );
 }
 
-export function recordRegistration( flow ) {
-	// Tracks
-	analytics.tracks.recordEvent( 'calypso_user_registration_complete', { flow } );
-	// Google Analytics
-	analytics.ga.recordEvent( 'Signup', 'calypso_user_registration_complete' );
-	// Marketing
-	adTrackRegistration();
-}
-
-export function recordPasswordlessRegistration( flow ) {
-	// Tracks
-	analytics.tracks.recordEvent( 'calypso_user_registration_passwordless_complete', { flow } );
-	// Google Analytics
-	analytics.ga.recordEvent( 'Signup', 'calypso_user_registration_passwordless_complete' );
-	// Marketing
-	adTrackRegistration();
-}
+export function recordRegistration( { userData, flow, type } ) {
```

Comment:
Nitpick: Could we add a JSDoc here explaining what's expected for each of the destructured data here? `flow` and `type` might be self-explanatory enough, but the shape of `userData` is hard to tell without looking up for a actual call.